common/djangoapps/student/models.py

```diff
@@ -974,7 +974,10 @@ def users_enrolled_in(self, course_id, include_inactive=False):
         }
         if not include_inactive:
             filter_kwargs['courseenrollment__is_active'] = True
-        return User.objects.filter(**filter_kwargs)
+        exclude_kwargs = {
+            'courseenrollment__user__email__endswith': 'example.com',
```

Comment:
This appears as if it would introduce a latent bug  where legitimate domains could be excluded, eg:
- contrived-example.com

To be thorough, should we instead check?:
```py
exclude_kwargs = {
            'courseenrollment__user__email__endswith': '.example.com',
            'courseenrollment__user__email__endswith': '@example.com',
}
```

This would exclude:
- user@example.com
- user@some.subdomain.example.com

while still including:
- user@contrived-example.com

Reply:
In thinking of submitting this upstream, I'm wondering if this exclusion should be optional, via a `kwarg` to this method, like `include_inactive=False`, but `exclude_fake_email=False`, to match the existing convention.

There may be times you legitimately want every single user account..